### DISCUSSION TITLE
Custom hyphenation configs

### DISCUSSION THREAD

**Discussion by bserem** [mundschenk-at/php-typography#158] [General]
I stumbled upon a problem with French language and this led me to the idea that it would be nice to allow custom values to be passed to the hyphenation.

Hyphenating this text: `Avec le Valais, on ne sait jamais à quoi s'attendre`
Results into this: `Avec le Valais, on né sait jamais à quoi s'attendre`
Notice the `né` vs the `ne`. A sign is added above the e.

My client says this is not correct (and since I do not remember my French I take his word for granted).

Is there any way to pass custom hyphenation configuration to the hyphenator?

same question posted: https://www.drupal.org/project/twig_typography/issues/2950828

**Comment by mundschenk-at** [comment]
That's not directly related to hyphenation, but rather to "Smart Diacritics". Currently only US English (the default) and German are supported for this setting. For other languages, you should disable this setting (and there may be false replacements even for those two, this is something you have to decide if you want to risk enabling it).

```php
$settings->set_smart_diacritics( false );
````
should fix the issue (assuming `$settings` is your `PHP_Typography\Settings` object.

**Comment by bserem** [comment]
I am using the php-typography project as a requirement for another Drupal module, so I need to find out where/how to pass these settings.

Can I somehow disable this for one language and keep it enabled for another?
(the project in question is in both German and French)

Thanks for your reply

**Comment by mundschenk-at** [comment]
Since you can have as many separate `Settings` objects as you like, in theory this should be no problem. However, I don't know how that Drupal module manages the settings, I can't tell you anything on how to do that in practice.

**Comment by mundschenk-at** [comment]
@bserem After thinking about this, I'd simply disable the setting even for German. It's main use is adding proper diacritics to some French loan words. This is something a compentent copy writer will do by themselves. Also, loan words tend to lose their accents as they become more integrated into the host language, so re-adding diacritics is a very normative approach.

Regarding the passing of options to the module: [See the "Configuration" section in the Twig Typography documentation](https://www.drupal.org/project/twig_typography). Apparently a YAML file called `typography_default.yml` should do the trick.

**Comment by bserem** [comment]
@mundschenk-at thanks, both for the definition of smart_diacritics (I didn't knew it) and for the support.

I am attempting to get these into the project. Will post back upon success!

**Comment by bserem** [comment]
As pointed out by mundschenk-at, adding this:
`set_smart_diacritics: FALSE`
to a file called `typography_default.yml` in the theme root folder did the trick.

Thanks for the support!
